pkg/cfg/sanitizer.go

```diff
@@ -0,0 +1,47 @@
+package cfg
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+)
+
+type Sanitizer func(in interface{}) (interface{}, error)
+
+func Sanitize(in map[string]interface{}, sanitizers []Sanitizer) (interface{}, error) {
+	var err error
+
+	if len(sanitizers) == 0 {
+		return in, nil
+	}
+
+	for key, val := range in {
+		for _, san := range sanitizers {
+			if in[key], err = san(val); err != nil {
```

Comment:
I suggest to not automically overwrite the map content, as in case of err the value returned from the sanitizer should not (must not) be trusted. This could lead to erasing contents.